File: 2018/01/python/part2.py
#!/usr/bin/env python3

from itertools import cycle

with open('../input', 'r') as file:
    frequency = 0
    seen = set()
    found = False
    for num in cycle(file.readlines()):
        frequency += int(num)
        if frequency in seen:
            print(frequency)
            break
        seen.add(frequency)

File: 2018/01/rust/src/bin/part_2.rs
use std::collections::HashSet;

fn main() {
    let mut frequency: i32 = 0;
    let mut seen = HashSet::new();
    for number in include_str!("../../../input")
        .lines()
        .map(|line| line.parse::<i32>().unwrap())
        .cycle()
    {
        frequency += number;
        if seen.contains(&frequency) {
            println!("{}", frequency);
            break;
        }
        seen.insert(frequency);
    }
}
